.github/workflows/ci.yml

```diff
@@ -9,18 +9,64 @@ on:
 
 jobs:
   noir:
-    name: Noir CI
+    name: lib
     runs-on: ubuntu-latest
     defaults:
       run:
-        working-directory: circuits
+        working-directory: lib
     steps:
       - uses: actions/checkout@v4
 
       - name: Install Nargo
         uses: noir-lang/noirup@v0.1.4
         with:
-          toolchain: v1.0.0-beta.11
+          toolchain: v1.0.0-beta.16
```

Comment:
Can we use v1.0.0-beta.15 here? It could help avoid surprises when migrating to enclave, which uses 15